### PR TITLE
[JENKINS-37324] Step descriptions proof of concept

### PR DESCRIPTION
**Pull request by svanoort**
Part of [JENKINS-37324](https://issues.jenkins-ci.org/browse/JENKINS-37324) and downstream of https://github.com/jenkinsci/workflow-cps-plugin/pull/98 plus its upstream dependencies.

TODO:
- [x] ~~Add sanitization of step description strings~~ **Automatically** handled by Handlebars escaping of content.

### CODE REVIEW

File: rest-api/src/main/java/com/cloudbees/workflow/rest/external/FlowNodeExt.java [hunk 1]
```diff
@@ -105,6 +105,15 @@ public ErrorExt getError() {
         return error;
     }
 
+    public void setParameterDescription(String desc) {
+        this.parameterDescription = desc;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getParameterDescription() {
```
Comment (rsandell) [inline rest-api/src/main/java/com/cloudbees/workflow/rest/external/FlowNodeExt.java]: Would be slightly less confused if plural was used, e.g. `getParametersDescription`

File: rest-api/src/test/java/com/cloudbees/workflow/rest/endpoints/JobAndRunAPITest.java [hunk 2]
```diff
@@ -37,13 +37,11 @@
 import hudson.model.Action;
 import hudson.model.Result;
 import hudson.model.queue.QueueTaskFuture;
-import jenkins.model.Jenkins;
 import org.jenkinsci.plugins.workflow.actions.TimingAction;
 import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
 import org.jenkinsci.plugins.workflow.graph.FlowNode;
 import org.jenkinsci.plugins.workflow.job.WorkflowJob;
 import org.jenkinsci.plugins.workflow.job.WorkflowRun;
-import org.jenkinsci.plugins.workflow.support.steps.StageStep;
```
Comment (jglick) [inline rest-api/src/test/java/com/cloudbees/workflow/rest/endpoints/JobAndRunAPITest.java]: unrelated

File: ui/src/main/js/view/templates/stage-logs.hbs [hunk 3]
```diff
@@ -5,6 +5,9 @@
         <div class="node-log-frame {{this.status}}" cbwf-controller="node-log" objectUrl="{{this._links.log.href}}">
             <div class="node-name"><span class="glyphicon glyphicon-collapse-down" title="Expand"></span><span class="glyphicon glyphicon-collapse-up" title="Collapse"></span>
                 {{this.name}}  {{#if this.durationMillis}}(self time {{formatTime this.durationMillis}}){{/if}}
+                {{#if this.parameterDescription}}
+                    - {{this.parameterDescription}}
```
Comment (jglick) [inline ui/src/main/js/view/templates/stage-logs.hbs]: `—` might look better.

Comment (svanoort) [inline ui/src/main/js/view/templates/stage-logs.hbs]: You're right, good point.  I switched to

"name -- argumentsString -- (self time 10 ms)"
I.e. 
"Shell Step -- rm -rf / -- (self time 10 hours)"

**Comment by svanoort** [comment]
@reviewbybees

File: ui/src/main/js/view/templates/stage-logs.hbs (hunk 3 above)
Comment (jglick) [inline ui/src/main/js/view/templates/stage-logs.hbs]: I actually was referring to an m-dash `—`, not a pair of ASCII hyphens `--`, but whatever. :-)

**Comment by bitwiseman** [comment]
@svanoort  I know this a POC, but are there any tests?

**Comment by svanoort** [comment]
@bitwiseman Good question.  Existing unit tests verify if this breaks anything for us - the net-new stuff is just getters/setters & a template operation -- not really any logic there to test.

I did manual testing to verify this works correctly in cases where we both do and do not have ArgumentsActions attached for steps, so am satisfied it works in integration.

**Pull request merged**